app/assets/javascripts/reducers/campaigns.js

```diff
@@ -33,7 +33,11 @@ const initialState = {
 
 export default function campaigns(state = initialState, action) {
   switch (action.type) {
-    case ADD_CAMPAIGN:
+    case ADD_CAMPAIGN: {
+      const newCampaign = action.data.course.campaigns;
+      const updatedCampaign = [...state.campaigns, newCampaign];
+      return { ...state, campaigns: updatedCampaign, isLoaded: true };
+    }
```

Comment:
What is the benefit of making this into a separate case?

Reply:
@ragesoss I had some misunderstandings regarding the definition and use of this reducer. but i have resolved and removed this separate case